### PR TITLE
bugfix: CLDSRV-361 fix exception with batch delete of null version

### PR DESCRIPTION
**Pull request by jonathan-gramain**
Fix an exception in Cloudserver when doing a batch delete containing a deletion of a null version that does not exist on a versioned object

### CODE REVIEW

**Comment by jonathan-gramain** [comment]
ping

File: lib/metadata/metadataUtils.js [hunk 1]
```diff
@@ -26,7 +26,13 @@ function getNullVersion(objMD, bucketName, objectKey, log, cb) {
         // the latest version is not the null version, but null version exists
         log.debug('null version exists, get the null version');
         options.versionId = objMD.nullVersionId;
-        return metadata.getObjectMD(bucketName, objectKey, options, log, cb);
+        return metadata.getObjectMD(bucketName, objectKey, options, log, (err, nullVersionMD) => {
+            if (err && err.is.NoSuchKey) {
```
Comment (tmacro) [inline lib/metadata/metadataUtils.js]: Not sure it its possible for it to be a non-Arsenal error, but better safe than sorry.
```suggestion
            if (err && err.is?.NoSuchKey) {
```

File: lib/metadata/metadataUtils.js [hunk 2]
```diff
@@ -96,6 +102,10 @@ function metadataGetObject(bucketName, objectKey, versionId, log, cb) {
     };
     return metadata.getObjectMD(bucketName, objectKey, options, log,
         (err, objMD) => {
+            if (err && err.is.NoSuchKey) {
```
Comment (tmacro) [inline lib/metadata/metadataUtils.js]:
```suggestion
            if (err && err.is?.NoSuchKey) {
```

**Review by KazToozs — APPROVED** [review]
LGTM

**Review by tmacro — APPROVED** [review]

**Pull request merged**